plugins/nodes/src/js/data/NetworkNodesClient.ts

```diff
@@ -0,0 +1,32 @@
+import { Observable } from "rxjs";
+import { request, RequestResponse } from "@dcos/http-service";
+import { map } from "rxjs/operators";
+
+export interface NetworkNode {
+  updated: string;
+  public_ips: string[];
+  private_ip: string;
+  hostname: string;
+}
+
+export type NetworkNodesResponse = NetworkNode[];
+
+export function fetchNetworkNodes(): Observable<RequestResponse<string[]>> {
+  return request("/net/v1/nodes").pipe(
+    map((reqResp: RequestResponse<any>) => {
+      if (reqResp.code >= 300) {
```

Comment:
Do you think we could make use of `partition` operator here? https://www.learnrxjs.io/operators/transformation/partition.html

Reply:
Yes we could the initial question which came to my mind is how does it help us if we split(partition) the stream and later we are merging it again. We are just increasing the number of function calls. The upside I see is that it would declutter the code and make it more readable. What do you think should we use the partition?

Reply:
Currently we're misusing map as we don't really map anything we just use it to filter errors and throw a side effect. I believe the upside of clear separation of intent is superior over a bit of extra performance.

Reply:
👍